epochStart/metachain/sovereignTrigger.go

```diff
@@ -9,22 +9,41 @@ import (
 	"github.com/multiversx/mx-chain-core-go/data/block"
 	"github.com/multiversx/mx-chain-go/dataRetriever"
 	"github.com/multiversx/mx-chain-go/epochStart"
+	"github.com/multiversx/mx-chain-go/process"
 )
 
+// ArgsSovereignTrigger defines args needed to create a sovereign trigger
+type ArgsSovereignTrigger struct {
+	*ArgsNewMetaEpochStartTrigger
+	PeerMiniBlocksSyncer process.ValidatorInfoSyncer
+}
+
 type sovereignTrigger struct {
 	*trigger
+	currentEpochValidatorInfoPool epochStart.ValidatorInfoCacher
+	peerMiniBlocksSyncer          process.ValidatorInfoSyncer
 }
 
 // NewSovereignTrigger creates a new sovereign epoch start trigger
-func NewSovereignTrigger(args *ArgsNewMetaEpochStartTrigger) (*sovereignTrigger, error) {
-	metaTrigger, err := newTrigger(args, &block.SovereignChainHeader{}, &sovereignTriggerRegistryCreator{}, dataRetriever.BlockHeaderUnit)
+func NewSovereignTrigger(args ArgsSovereignTrigger) (*sovereignTrigger, error) {
+	if check.IfNil(args.PeerMiniBlocksSyncer) {
+		return nil, epochStart.ErrNilValidatorInfoProcessor
```

Comment:
error should be with PeerMiniBlocksSyncer

Reply:
Copy-pasted `PeerMiniBlocksSyncer` variable name from the other trigger, but it's actually a `ValidatorInfoSyncer`. Renamed variable

Reply:
ok